### PR TITLE
www-misc/Buku: New package

### PR DESCRIPTION
**Pull request by stkw0**
Buku is a popular command-line bookmark manager
Other distros like debian already have a package for it

Bug: https://bugs.gentoo.org/640612

### CODE REVIEW

**Comment by radhermit** [comment]
Unless the executable name is capitalized I'd make the package name lowercase and perhaps pull the tarball from pypi instead of github. In addition, the test phase should be run and required test deps added unless it blows in half or does terrible things. :smile:

**Comment by stkw0** [comment]
Thanks for the review. Initially I used the capitalized version to avoid changing the build directory name (the decompressed directory name is capitalized). 

Tests are also executed, but seems that are failing, so I still have to fix it. I will try to run all important tests only (without pylint nor pep8 tests).

**Comment by radhermit** [comment]
The decompressed dir name isn't capitalized on the prepared tarball via pypi. Generally I just use pypi tarballs for python projects instead of raw github tarballs if upstream provides them and they provide all relevant data that we'd want access to in ebuilds, e.g. tests/test data.

**Comment by stkw0** [comment]
Yes, seems that I got confused. 

I fixed the tests and changed SRC_URI.

**Comment by stkw0** [comment]
I fixed the installation of the program itself. It should be good enough now.

**Comment by stkw0** [comment]
Any news?

File: www-misc/buku/buku-3.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (titanofold) [inline www-misc/buku/buku-3.5.ebuild]: We're in 2018 now.

File: www-misc/buku/buku-3.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python3_{4,5,6} )
+
+inherit distutils-r1 bash-completion-r1
```
Comment (titanofold) [inline www-misc/buku/buku-3.5.ebuild]: Alphabetical, please.

File: www-misc/buku/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>davidroman96@gmail.com</email>
+		<name>David Roman</name>
+	</maintainer>
+	<upstream>
```
Comment (titanofold) [inline www-misc/buku/metadata.xml]: Add proxy project to maintainer list.

File: www-misc/buku/buku-3.5.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytest-catchlog[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		dev-python/py[${PYTHON_USEDEP}]
+		dev-python/flake8[${PYTHON_USEDEP}]
+	)
+"
+
+python_install() {
```
Comment (titanofold) [inline www-misc/buku/buku-3.5.ebuild]: I'm not sure if this is right. I think it should by python_install_all() that you override.

File: www-misc/buku/buku-3.5.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	dev-python/cryptography[${PYTHON_USEDEP}]
+"
+
+DEPEND="${RDEPEND}
+	test? (
+		dev-python/pytest[${PYTHON_USEDEP}]
+		dev-python/pytest-catchlog[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		dev-python/py[${PYTHON_USEDEP}]
+		dev-python/flake8[${PYTHON_USEDEP}]
```
Comment (titanofold) [inline www-misc/buku/buku-3.5.ebuild]: Alphabetical, please.

**Comment by stkw0** [comment]
@titanofold All done

**Pull request closed without merging**